src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Windows.cs

```diff
@@ -1,13 +1,77 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Diagnostics;
+
 namespace System.Runtime.InteropServices
 {
-    public static class RuntimeInformation
+    public static partial class RuntimeInformation
     {
+        private static string s_osDebugName = null;
+        private static Architecture? s_osArch = null;
+
         public static bool IsOSPlatform(OSPlatform osPlatform)
         {
             return OSPlatform.Windows == osPlatform;
         }
+
+        public static string GetOSDebugName()
+        {
+            if (null == s_osDebugName)
+            {
+                string type = Interop.mincore.IsWindowsServer() ? "Server" : "Client";
+                string version = "Unknown";
+
+                if (Interop.mincore.IsWindows10OrGreater())
+                    version = "10";
```

Comment:
From my experience the Version family functions (i.e. GetVersion, VerifyVersion, IsWinXOrGreater) all work the same way and may not return the expected version unless the application is manifested. Interestingly, GetVersionEx returns 10.0 on Nano server. I don't remember from the top of my head what GetVersionEx returns on Win10 Iot Core (read: Windows 10 on Raspberry Pi2) but I _think_ it was 10 as well. I believe this is important when an app P/Invokes to a native dll. If the version returned by the OS is 10 the dll used by the app should be linked against onecore, otherwise it should be linked against kernel32. 

Note that from this point of view having the `win81` RID may not make sense because without a manifest you will never be able to detect this. `win10` RID might be useful to detect OneCore. /cc @ericstj